platform/macos/os_macos.mm

```diff
@@ -820,8 +823,8 @@
 	// Implicitly create shared NSApplication instance.
 	[GodotApplication sharedApplication];
 
-	// In case we are unbundled, make us a proper UI application.
```

Comment:
This will cause a dock icon to appear, then disappear again and reappear when the window is opened (if the icon is not pinned to dock).

Reply:
Got it. I will remove this and change the macos_tools plist to contain the LSUIElement setting.